### PR TITLE
XS✔ ◾ Adding Front Door vs Cloudflare comparison as CDNs 

### PR DESCRIPTION
**Pull request by amankumarrr**
>
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

✏️ We are moving our name servers and redirects from Front Door to Cloudflare

> 2. What was changed?

✏️ Added comparison for Cloudflare and Front Door 

> 3. Did you do pair or mob programming (list names)?

✏️ Yes, I worked with Brady to get it reviewed first

### CODE REVIEW

File: rules/do-you-use-cdn-for-js-files/rule.md [hunk 1]
```diff
@@ -36,23 +38,57 @@ CDNs not only ensure a faster experience to your users, but they also help to pr
...
+* **Limited Redirects**: Front Door restricts bulk redirects with a [limit of 800 resources](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/azure-subscription-service-limits) per resource group.
+* **Time-Consuming Deployments**: Deployments using Bicep can take a long time, sometimes up to hours.
+* **Lack of Flexibility**: It doesn’t offer much flexibility for custom redirects, such as using regular expressions.
+* **Painful Cache Purging**: Cache purging can be unreliable and can take up to 15 minutes.
+* **High Costs**: It is quite expensive compared to other CDNs.
+
+✅ Pros:
+
+* **Easy Integration**: Seamlessly integrates with other Azure services.
+* **Flexible Origin Handling**: Better handling for origin groups on a route level (e.g., /people can be hosted on a different server).
```
Comment (bradystroud) [inline rules/do-you-use-cdn-for-js-files/rule.md:58]: We usually start with the pros

File: rules/do-you-use-cdn-for-js-files/rule.md [hunk 2]
```diff
@@ -36,23 +38,57 @@ CDNs not only ensure a faster experience to your users, but they also help to pr
...
+Cloudflare is known for its Distributed Denial-of-Service (DDoS) protection and Web Application Firewall (WAF), along with a host of other options.
+
+✅ Pros:
+
+* **Easy Deployment**: Simple deployment through Wrangler. (it takes a few seconds to deploy workers)
+* **Flexible Redirects**: More flexible in handling bulk and customizable redirects using [Cloudflare Workers](https://developers.cloudflare.com/workers/).
+* **Enhanced Security**: Provides better WAF protection and secure DDoS mitigation.
+* **Cost-Effective**: Much cheaper than other CDNs.
+* **Efficient Caching**: Better caching options.
+* **Instant Cache Purging**: Offers an instant purging option for caches.
```
Comment (bradystroud) [inline rules/do-you-use-cdn-for-js-files/rule.md:73]: No cons for Cloudflare?

File: rules/do-you-use-cdn-for-js-files/rule.md [hunk 3]
```diff
@@ -36,23 +38,57 @@ CDNs not only ensure a faster experience to your users, but they also help to pr
...
+6. <https://www.cloudflare.com/network/china/> (Cloudflare’s China Service)
+7. <https://www.akamai.com>
+8. <https://cdnjs.cloudflare.com>
 
 #### Which ones do not work well from China?
 
-1. https://maxcdn.bootstrapcdn.com
-2. https://ajax.googleapis.com
+1. <https://maxcdn.bootstrapcdn.com>
+2. <https://ajax.googleapis.com>
```
Comment (bradystroud) [inline rules/do-you-use-cdn-for-js-files/rule.md:91]: I think one of these should be recommended (maybe the Cloudflare one)

**Review by bradystroud — CHANGES_REQUESTED** [review]
looking good

File: rules/do-you-use-cdn-for-js-files/rule.md (hunk 1 above)
Comment (amankumarrr) [reply rules/do-you-use-cdn-for-js-files/rule.md:58]: done

**Review by bradystroud — APPROVED** [review]

**Pull request merged**